tests/pgsmo_tests/utils.py

```diff
@@ -106,7 +108,8 @@ def _execute(self, query, params=None):
         self._has_been_read = True
 
 
-class MockConnection(connection):
+class MockConnection():
```

Comment:
Given that this class is a mock for our PGConnection we should use that as a base class. Then you would not have to define all these redundant properties and methods and get them from the PGConnection class. This way we can mock what we need for a particular test using patch call. Currently what is happening that we are not testing our product code but testing the test code. Right now if we create a new MockConnection, it does not even call the __init__ method. If we ever make a change to that init method our tests won't even catch it as we are not testing that code.